### DISCUSSION TITLE
Nested templates don't work properly

### DISCUSSION THREAD

**Discussion by everydayanchovies** [minad/tempel#133] [Q&A]
When expanding a template while filling in another template, the y in the form (p x y) no longer updates.

Consider these two templates:

`(template-a "abc")
(template-b (p "" var) n (format "%s" var))`

When first expanding template b and then expanding template a within template b, I expect 

"abc
abc", 

but instead I get 

"abc"

**Comment by everydayanchovies** [comment]
More precisely, template-a is inserted after the region which is watched to replace the value of y.

**Reply by minad** [reply]
I see your point. I think Tempel doesn't support nested fields right now. I don't know how hard it would be to lift this restriction. I doubt that the added complexity would be worth it.

**Reply by everydayanchovies** [reply]
I see, thanks. The use case for me is coupling calc with org mode latex preview. This way I can simplify latex expressions as they are written.
